### PR TITLE
Skip Apple Pay registration for accounts from India

### LINKED ISSUE #2566: Prevent Apple Pay domain registration for Indian merchants
Context: p1677094614234679-slack-C9976E5MJ

### Description

As Apple Pay is not supported in India, we should stop invoking domain registration requests for Indian merchants. This should be relatively straightforward and should reduce domain registration errors.

### Acceptance criteria

Stop invoking Apple Pay domain registration requests for merchants based in India.

### Dev notes

As @reykjalin suggests [here](p1677873913689259-slack-CGGCLBN58), we can probably check the store address, currency, or perhaps the Stripe account country to prevent these requests.

### PR DESCRIPTION
**Pull request by a-danae**
Fixes #2566

## Changes proposed in this Pull Request:

- Bail out before triggering Apple Pay registration if the account country is India.

## Testing instructions

**Confirming the registration is triggered with a supported account**
1. Connect the Stripe plugin to a test account, not from India. The one you probably have already would do
2. Add some fake live keys (the domain registration isn't triggered if they're empty). Stripe plugin Settings page `{site}/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings` -> General -> Edit account keys -> Live -> Enter `pk_live_1234` for the live key and `sk_live_1234` for the secret key
3. Ensure test mode is enabled. Stripe plugin Settings page -> General -> Enable test mode
4. Ensure logging is enabled. Stripe plugin Settings page -> Advanced settings -> Check "Log error messages"
5. Disable Apple Pay / Google Pay Express checkouts, save, enable them again, and save. Stripe plugin Payment Methods page `{site}/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=methods` -> Express checkouts -> Apple Pay / Google Pay
6. Go to Stripe's log file. `/wp-content/uploads/wc-logs/woocommerce-gateway-stripe-{date}.log`
7. Confirm there's an entry with the following
```
====Start Log====
Error: Unable to verify domain - Invalid API Key provided: sk_live_1234
====End Log====
```

**Confirming the registration isn't triggered with an account from India**
1. Create a test Stripe account with India as its country (or ping me for some creds)
2. Connect the Stripe plugin to this account. Stripe plugin Settings page `{site}/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings` -> General -> Edit account keys -> Test -> Enter the keys
3. Disable Apple Pay / Google Pay Express checkouts, save, enable them again, and save. Stripe plugin Payment Methods page `{site}/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=methods` -> Express checkouts -> Apple Pay / Google Pay
4. Go to Stripe's log file. `/wp-content/uploads/wc-logs/woocommerce-gateway-stripe-{date}.log`
5. Confirm no entry for domain verification was added

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [x] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)

### CODE REVIEW

File: includes/class-wc-stripe-apple-pay-registration.php [hunk 1]
```diff
@@ -412,6 +416,22 @@ public function admin_notices() {
...
+
+	/**
+	 * Returns whether Apple Pay is available in the account country.
+	 *
+	 * @since 7.6.0
+	 *
+	 * @return boolean
+	 */
+	private function is_available_in_account_country(): bool {
+		$cached_account_data = WC_Stripe::get_instance()->account->get_cached_account_data();
```
Comment (a-danae) [inline includes/class-wc-stripe-apple-pay-registration.php:428]: I tried injecting `WC_Stripe::get_instance()->account` as a dependency of this class, WC_Stripe_Apple_Pay_Registration, but some unit tests started to fail for WC_REST_Stripe_Settings_Controller for some reason. I debugged it for a bit but preferred not to spend too much time there as this approach is used in other places too. All good with updating it if convenient.

File: includes/class-wc-stripe-apple-pay-registration.php [hunk 2]
```diff
@@ -412,6 +416,22 @@ public function admin_notices() {
...
 	}
+
+	/**
+	 * Returns whether Apple Pay is available in the account country.
+	 *
+	 * @since 7.6.0
+	 *
+	 * @return boolean
+	 */
+	private function is_available_in_account_country(): bool {
```
Comment (mattallan) [inline includes/class-wc-stripe-apple-pay-registration.php]: What are your thoughts on renaming this new function to be a bit more generic i.e. `is_available()` or `is_supported()`, or `can_register_domain()`?

I was thinking in the future there may be more reasons we need to restrict stores registering with Apple Pay like currency or something else tied to their stripe account.

Just a passing thought, happy with this function as is.

**Review by mattallan — APPROVED** [review]
Nice changes @a-danae ! I've tested this PR with a Stripe account from India and confirmed it doesn't attempt to register the domain with Apple Pay and then connected my usual US account and confirmed the Apple Pay domain registration code was run.

I only left a small comment about a function name but not blocking IMO :)

Approving

File: includes/class-wc-stripe-apple-pay-registration.php (hunk 1 above)
Comment (mattallan) [reply includes/class-wc-stripe-apple-pay-registration.php:428]: I'm still not overly familiar with what the best practices are when working on the Stripe code but this seems totally fine to me 😅

File: includes/class-wc-stripe-apple-pay-registration.php (hunk 2 above)
Comment (a-danae) [reply includes/class-wc-stripe-apple-pay-registration.php]: Makes complete sense to me. Thanks for that observation, Matt! Updated in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2676/commits/d92a068af2a87034d7410fab331adbc60a8895c5

**Pull request merged**